docs/academy/08-AdvancedTopics/02-Reactor/intro.md

```diff
@@ -0,0 +1,614 @@
+---
+sidebar_position: 0
+---
+
+# Getting Started with Reactor
+
+## Introduction
+
+Welcome to the Powerhouse Reactor system (aka document-drive). Reactor is a powerful document management and synchronization engine designed to handle collaborative workflows across distributed systems. As a core component of the Powerhouse platform, Reactor provides robust capabilities for document creation, storage, versioning, and real-time collaboration.
+
+Reactor enables applications to work with structured documents in a collaborative and reactive way. It maintains document state through a series of operations, supports versioning, and handles synchronization across multiple clients and servers.
+
+## Documents
+
+The Reactor system is built around the concept of documents. Documents are the core data structure that Reactor manages, and they can be organized hierarchically. Documents that hold other documents are called **drives**.
+
+> Note (Prometheus): Simplify
+
+![Document Management](./images/document-management.svg)
+
+**Operations** are the mechanism by which documents are modified. When a client wants to change a document, it sends operations to the Reactor, which applies them to the document.
+
+![Operations Input](./images/operations-input.svg)
+
+When a client requests a document, the Reactor reconstructs it by applying all of its operations in sequence, resulting in the current state of the document.
+
+![Document Reconstruction](./images/document-reconstruction.svg)
+
+## Document Models
+
+Document models are structured descriptions that define both the shape of a document and the operations that can be applied to it. They serve as blueprints for creating and manipulating documents within the Reactor system.
+
+Each document model specifies:
+- The document's data schema: fields and types
+- Valid operations that can modify the document
+- Rules for how operations transform the document state
+
+Document models in the Reactor system are themselves documents. This self-describing pxroperty allows document models to be stored, versioned, and synchronized using the same mechanisms as any other document in the system.
+
+When creating a new document, you must specify which document model it follows. The Reactor then ensures that all operations applied to that document conform to the rules defined by its model.
+
+> Note (Prometheus): Document Model is generated from a Document Model Specification
+
+![Operation Validation](./images/operation-validation.svg)
+
+## Data Objects
+
+Reactor is built around a well-defined data model that represents documents, operations, and the relationships between them.
+
+![Entity Relationship](./images/entity-relationships.svg)
+
+#### PHDocument
+
+The core entity representing a **document**. Each document:
+
+- Has a unique `id`
+- Contains `revision` information
+- Maintains a collection of "_scopes_" (more on this later)
+- Stores a specific `documentType` that defines its structure
```

Comment:
We discussed adding the current document model version to the document.
Should we mention `documentTypeVersion` or `version` here?
Or omit it here, to reduce the information overload, and refer to it later on a specific `Document Model Versioning` topic.

Reply:
Likely a good idea to have both:

`version` would refer to the `PHDocument` structure itself.
`documentTypeVersion`would refer to the relevant document model type.